examples/scripts/learning_regularization_parameter_maps_for_tv_reconstruction.py

```diff
@@ -0,0 +1,876 @@
+        )
+        self.gradient_operator = gradient_operator
+
+        self.g = mrpro.operators.functionals.ZeroFunctional()
+
+        # operator norm of the stacked operator K=[A, \nabla]^T
+        self.stacked_operator_norm = 3.0  # analytically calculated
+
+    def estimate_lambda_map(self, image: torch.Tensor) -> torch.Tensor:
+        """Estimate regularization parameter map from image."""
+        # squeeze the dimensions that are one. In particular, the initial image
+        # is a coil-combined image (i.e. coils=1) and because, we only consider 2D problems here,
+        # z can be assumed to be always 1.
+        # (other*, coils, z, y, x) -> (other*, y, x)
+        input_image = image.abs().squeeze(-3)
+        regularization_parameter_map = self.lambda_map_network(input_image).unsqueeze(-3).unsqueeze(-3)
+        return regularization_parameter_map
+
+    def forward(
+        self,
+        initial_image: torch.Tensor,
+        csm: torch.Tensor,
+        kdata: torch.Tensor,
+        traj: mrpro.data.KTrajectory,
+        regularization_parameter: torch.Tensor | None = None,
+    ) -> torch.Tensor:
+        """Reconstruct image using an unrolled PDHG algorithm.
+
+        Parameters
+        ----------
+        initial_image
+            initial guess of the solution of the TV problem.
+        csm
+            coil sensitivity map tensor of the considered problem.
+        kdata
+            k-space data tensor of the considered problem.
+        traj
+            k-space trajectory tensor of the Fourier considered operator.
+        regularization_parameter
+            regularization parameter to be used in the TV-functional. If set to None,
+            it is estimated by the lambda_map_network.
+            (can also be a single scalar)
+
+        Returns
+        -------
+            Image reconstructed by the TV-minimization algorithm.
+        """
+        # if no regularization parameter map is provided, compute it with the network
+        if regularization_parameter is None:
+            regularization_parameter = self.estimate_lambda_map(initial_image)
+
+        f_1 = 0.5 * mrpro.operators.functionals.L2NormSquared(target=kdata, divide_by_n=False)
+        f_2 = mrpro.operators.functionals.L1NormViewAsReal(weight=regularization_parameter, divide_by_n=False)
+        f = mrpro.operators.ProximableFunctionalSeparableSum(f_1, f_2)
+
+        fourier_operator = mrpro.operators.FourierOp(
+            recon_matrix=self.img_shape,
+            encoding_matrix=self.k_shape,
+            traj=traj,
+        )
+        csm_operator = mrpro.operators.SensitivityOp(csm)
+        forward_operator = fourier_operator @ csm_operator
+        stacked_operator = mrpro.operators.LinearOperatorMatrix(((forward_operator,), (self.gradient_operator,)))
+
+        primal_stepsize = dual_stepsize = 0.95 * 1.0 / self.stacked_operator_norm
```

Comment:
if we dont calculate the operator_norm, please instead just set the primal and dual stepsize in the init.